sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsConf.scala

```diff
@@ -1891,48 +1916,88 @@ object RapidsConf {
     println("-----|-----------------|-------------|---------------|------")
   }
 
-  def help(asTable: Boolean = false): Unit = {
+  def help(asTable: Boolean = false, commonlyUsed: Option[Boolean] = None): Unit = {
```

Comment:
I would create two `help` methods to print configs for common ones and advanced ones separately.
e.g. 
`helpCommonConfs`, `helpAdvancedConfs` to avoid deeper if-else blocks, which is not straightforward to know how the advanced page looks like.

And seems `commonlyUsed` being a boolean will be enough, no need to have it being a Option. Since there is no place to call this `help` with `commonlyUsed` being a None.

Reply:
User may call `RapidsConf.help()` in Spark shell like [doc here](https://github.com/NVIDIA/spark-rapids/tree/branch-23.06/docs/dev#generating-the-configuration-documentation), so I set `commonlyUsed` default to None to let it print all config by default. Creating two more help methods can avoid this problem, but will generate many redundant codes because we still want the user to be able to call help directly in the spark shell.

Reply:
Done, thanks